dpytools/http/upload.py

```diff
@@ -0,0 +1,233 @@
+        print("Upload to s3 complete")
+
+        return s3_key, s3_uri
+
+    def _upload_file_chunks(
+        self, file_chunks: list[str], upload_params: dict, florence_access_token: str
+    ) -> None:
+        """
+        Upload file chunks to DP Upload Service with the specified upload parameters.
+        """
+        chunk_number = 1
+        for file_chunk in file_chunks:
+            with open(file_chunk, "rb") as f:
+                # Load file chunk as binary data
+                file = {"file": f}
+
+                # Add chunk number to upload request params
+                upload_params["resumableChunkNumber"] = chunk_number
+
+                # Submit `POST` request to `self.upload_url`
+                self.post(
+                    url=self.upload_url,
+                    headers={"X-Florence-Token": florence_access_token},
+                    params=upload_params,
+                    files=file,
+                    verify=True,
+                )
+                # TODO Replace print statements with logging
+                print(f"File chunk {chunk_number} of {len(file_chunks)} posted")
+                chunk_number += 1
+
+
+def _generate_upload_params(csv_path: Union[Path, str], chunk_size: int) -> dict:
+    """
+    Generate request parameters that do not change when iterating through the list of file chunks.
+
+    To be used with the `upload` endpoint.
+    """
+    # Get total size of file to be uploaded
+    total_size = os.path.getsize(csv_path)
+
+    # Convert csv_path string to Path
+    if not isinstance(csv_path, Path):
+        csv_path = Path(csv_path).absolute()
+
+    # Get filename from csv filepath
+    filename = str(csv_path).split("/")[-1]
+
+    # Get timestamp to create `resumableIdentifier` value in `POST` params
+    timestamp = datetime.datetime.now().strftime("%d%m%y%H%M%S")
+
+    # Generate upload request params
+    upload_params = {
+        "resumableTotalChunks": ceil(total_size / chunk_size),
+        "resumableChunkSize": chunk_size,
+        "resumableTotalSize": total_size,
+        "resumableType": "text/csv",
+        "resumableIdentifier": f"{timestamp}-{filename.replace('.', '-')}",
+        "resumableFilename": filename,
+    }
+    return upload_params
+
+
+def _generate_upload_new_params(
+    csv_path: Union[Path, str],
+    s3_path: str,
+    title: str,
+    is_publishable: bool = False,
+    collection_id: str = "",
+    licence: str = "Open Government Licence v3.0",
+    licence_url: str = "http://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/",
+) -> dict:
+    """
+    Generate request parameters that do not change when iterating through the list of file chunks.
+
+    To be used with the `upload-new` endpoint.
+    """
+    # Get total size of file to be uploaded
+    total_size = os.path.getsize(csv_path)
+
+    if ceil(total_size / 5242880) > 10000:
+        raise ValueError("File too big")
```

Comment:
very well worked out...but....whats the actual behaviour if we exceed it?

it the api responds with any variation of "file too big" then we don't need to do this and it might be detrimental. I'm thinking of a scenario where that limit on the api just happens to be changed or upped (which seems feasible) but we cant use it because our client is hard coded to old constraints.

we can enforce this if we have to, but im be leary of doing so unless we really, really have to. If the api has constraints it should be on the api to say so if we breach them.